native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java

```diff
@@ -390,8 +391,9 @@ private static BObject createRequest(HttpCarbonMessage httpCarbonMessage, BObjec
 
     static BObject getCaller(Resource resource, HttpCarbonMessage httpCarbonMessage,
                              BMap<BString, Object> endpointConfig) {
+        String resourceAccessor = resource.getBalResource().getAccessor().toUpperCase(Locale.getDefault());
         final BObject httpCaller = httpCarbonMessage.getProperty(HttpConstants.CALLER) == null ?
-                ValueCreatorUtils.createCallerObject(httpCarbonMessage) :
+                ValueCreatorUtils.createCallerObject(httpCarbonMessage, resourceAccessor) :
```

Comment:
With the interceptor implementation we have shared a single caller object among many services. So in this function we trying to get the shared caller and reusing it. 
Suppose we have created the caller for the `default` resource in the interceptor and when we go the actual resource which is a `POST` one, then we reuse the same caller. That means the resource accessor in the caller is `default` but the actual response is from a `POST` resource. This will return a 200 OK rather than 201 Created.
So we need to update the accessor field in the caller, every time we call it in different services.
This should be done for `RequestInterceptoUnitCallback` as well

Reply:
`http:Caller` is associated with `http:Request`. Then again `http:Request` is associated with a given HTTP resource. Hence, `http:Caller` should be associated with an HTTP resource. As per my understanding `http:Caller` will be created per resource (Original resource. - according to your comment it would be a `POST` resource) in out current implementation [1]. So, no need to update the `resourceAccessor` in `RequestInterceptoUnitCallback` level. 

PS: If we are sharing the same caller object among multiple interceptors resources which are not associated with the current request flow, the basis of the implementation is broken logically and it violates the associations mentioned above. 

[1] - [https://github.com/ballerina-platform/module-ballerina-http/blob/760807caddc706a3193b528368cefe71cfd59329/native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java#L282](https://github.com/ballerina-platform/module-ballerina-http/blob/760807caddc706a3193b528368cefe71cfd59329/native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java#L282)

Reply:
Fixed the issue with [e35e4b5e608e9d17f82d99a6dbad62df6a8f3e6b](https://github.com/ballerina-platform/module-ballerina-http/pull/1101/commits/e35e4b5e608e9d17f82d99a6dbad62df6a8f3e6b)